reconcile/utils/glitchtip/client.py

```diff
@@ -0,0 +1,196 @@
+from typing import Any, Optional
+from urllib.parse import urljoin
+
+import requests
+from reconcile.utils.glitchtip.models import Organization, Project, Team, User
+from sretoolbox.utils import retry
+
+
+def get_next_url(links: dict[str, dict[str, str]]) -> Optional[str]:
+    """Parse glitchtip's response header 'Link' attribute and return the next page url if exists.
+
+    See
+    * https://gitlab.com/glitchtip/glitchtip-backend/-/blob/master/glitchtip/pagination.py#L34
+    * https://requests.readthedocs.io/en/latest/api/?highlight=links#requests.Response.links
+    """
+    if links.get("next", {}).get("results", "false") == "true":
+        return links["next"]["url"]
+    return None
+
+
+class GlitchtipClient:
+    def __init__(self, host: str, token: str) -> None:
+        self.host = host
+        # todo timeout
+        self._session = requests.Session()
+        self._session.headers.update(
+            {
+                "Authorization": f"Bearer {token}",
+                "Content-Type": "application/json",
+            }
+        )
+
+    @retry()
```

Comment:
Did you think about using a Custom HTTPAdatapter with a Retry setting? That way you could make the retries configurable in the client.

Reply:
I didn't know about custom HTTPAdapter; looks promising! I'll give it a try.